torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -0,0 +1,157 @@
+#pragma once
+
+#include "CUDAUtils.hpp"
+#include "NCCLUtils.hpp"
+#include "ProcessGroup.hpp"
+#include "Store.hpp"
+
+// forward declaration
+struct THCState;
+
+namespace c10d {
+
+// ProcessGroupNCCL implements NCCL bindings for c10d.
+//
+// All functions of the class are expected to be called in the same order
+// across all processes in the process group.  This is the only way that we
+// can guarantee to match up the same calls among all processes.
+//
+// All NCCL functions provided by this class are asynchronous functions. More
+// specifically, each NCCL call is scheduled on a separate CUDA stream that is
+// different from the current THC CUDA stream. This is for the purpose of
+// achieving potentially concurrency and better performance. As a result,
+// it is the caller's responsibilty to make sure that the CUDA stream his/her
```

Comment:
I like to use their >> his/her so it doesn't distract from the flow of the sentence